packages/builder/src/components/design/settings/controls/ColumnEditor/ColumnEditor.svelte

```diff
@@ -21,9 +22,14 @@
 
   $: datasource = getDatasourceForProvider($currentAsset, componentInstance)
   $: schema = getSchema($currentAsset, datasource)
-  $: options = Object.keys(schema || {})
+  $: options = allowCellEditing
+    ? Object.keys(schema || {})
+    : enrichedSchemaFields?.map(field => field.name)
   $: sanitisedValue = getValidColumns(value, options)
   $: updateBoundValue(sanitisedValue)
+  $: enrichedSchemaFields = getFields(Object.values(schema) || [], {
+    allowLinks: true,
```

Comment:
Should this check whether the table is SQL or not? If not SQL then we cannot allow links (not filterable for internal table).

Reply:
I did test this scenario, and related fields do not appear for internal tables.

It is actually already accounted for in the `getFields` function:

```js
if (!table || !table.sql) {
    return []
  }
```

See `searchFields.js`.

Reply:
ah - is that not the `getTableFields` function? as the `getFields` function isn't aware of the table - its just the schema.

Would it be beneficial to change this to its table counterpart rather than using the raw `getFields` to avail of this functionality?

Reply:
As discussed, `getFields` calls `getTableFields` so it should be fine as is!